src/API/Instrumentation/AutoInstrumentation/ExtensionHookManager.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\API\Instrumentation\AutoInstrumentation;
+
+use function assert;
+use Closure;
+use function extension_loaded;
+use Nevay\SPI\ServiceProviderDependency\ExtensionDependency;
+use OpenTelemetry\Context\Context;
+use OpenTelemetry\Context\ContextInterface;
+use OpenTelemetry\Context\ContextKeyInterface;
+use ReflectionFunction;
+
+/** @phan-file-suppress PhanUndeclaredClassAttribute */
+
+#[ExtensionDependency('opentelemetry', '^1.0')]
```

Comment:
@Nevay I don't think SPI is checking this. If I disable opentelemetry extension, I'd expect to not get this back from `ServiceLoader::load(HookManager::class)`, but I do.

Reply:
Dependencies are only checked during autoload dump in `v0.2.0`.
`v0.2.1` revalidates extension dependencies during runtime -> should work without `composer dump-autoload` after updating to the newest version.

Reply:
tested against 0.2.1, and it works as expected.